src/Query/AggregationBuilder.php

```diff
@@ -0,0 +1,98 @@
+<?php
+
+declare(strict_types=1);
+
+namespace MongoDB\Laravel\Query;
+
+use Illuminate\Support\Collection as LaravelCollection;
+use Illuminate\Support\LazyCollection;
+use InvalidArgumentException;
+use Iterator;
+use MongoDB\Builder\BuilderEncoder;
+use MongoDB\Builder\Stage\FluentFactoryTrait;
+use MongoDB\Collection as MongoDBCollection;
+use MongoDB\Driver\CursorInterface;
+use MongoDB\Laravel\Collection as LaravelMongoDBCollection;
+
+use function array_replace;
+use function collect;
+use function sprintf;
+use function str_starts_with;
+
+class AggregationBuilder
+{
+    use FluentFactoryTrait;
+
+    public function __construct(
+        private MongoDBCollection|LaravelMongoDBCollection $collection,
```

Comment:
Is there any reason to allow `MongoDB\Collection` here? It's only constructed as such in the test suite, but I presume you could just mock `MongoDB\Laravel\Collection` instead there.

I do realize `MongoDB\Laravel\Collection` just forwards method calls to the PHPLIB class, so there's no technical reason that a `MongoDB\Collection` wouldn't work as well.